src/registriesView.ts

```diff
@@ -97,6 +102,48 @@ export class ComponentTypesView implements TreeDataProvider<ComponentType> {
         return this.registries;
     }
 
+    public getCompDescriptions(): Set<ComponentTypeDescription> {
+        return this.compDescriptions;
+    }
+
+    public getListOfRegistries(): Registry[] {
+        return this.registries;
+    }
+
+    public async getAllComponents(): Promise<boolean> {
+        return new Promise<boolean>((resolve) => {
+            const compDescs: Set<ComponentTypeDescription> = new Set<ComponentTypeDescription>();
+            void getInstance().getCompTypesJson().then(async (devFileComponentTypes: DevfileComponentType[]) => {
+                const components = new Set<string>();
```

Comment:
no need to call getComponentTypesOfJSON this is old function introduced to deal with two king of component types s2i and devfile. Now there is only one king of components - defvile.